docs/usage.rst

```diff
@@ -81,13 +81,28 @@ To activate or source an environment, use the ``fzirob change_environment
 ENV_NAME`` command. This command sources the appropriate setup scripts of the
 environment depending on its contents.
 
+
 You can use tab completion on the environments so typing ``fzirob
 change_environment`` and then pressing :kbd:`Tab` should list all environments
 present.
 
 Executing ``fzirob change_environment`` without any environment specified will
 source the most recently sourced environment.
 
+
+Adding custom source commands
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition to the environment's workspace setup files, each environment contains a
+``setup_local.sh`` file that will get sourced during the ``change_environment`` call.
+
+.. note::
+
+   Using ``fzirob change_environment`` will reset your shell environment before sourcing a
```

Comment:
It might be helpful to mention here that it will reset to the state before robot_folders are sourced in the .bashrc/.zshrc. This could also help with confusion that may arise if things are sourced after robot_folders in the .bashrc/.zshrc and are then not present after a `fzirob change_environment`.

Reply:
Thanks, I rephrased that a little bit.